Upstream-Status: Backport

https://crrev.com/c/2246170

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 18b58ad84824224063b9373f9a6bed2a9e095567 Mon Sep 17 00:00:00 2001
From: Kristian Kirs <kristian.kirs@lge.com>
Date: Mon, 15 Jun 2020 19:08:54 +0000
Subject: [PATCH] GCC: Fix DHECK_EQ

GCC complains about it

Change-Id: I4c5ffcc66225c895efde77a77c83342d6794b5e1
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2246170
Reviewed-by: Christian Biesinger <cbiesinger@chromium.org>
Reviewed-by: Koji Ishii <kojii@chromium.org>
Commit-Queue: Koji Ishii <kojii@chromium.org>
Cr-Commit-Position: refs/heads/master@{#778406}
---
 .../blink/renderer/core/layout/ng/inline/ng_inline_node.cc      | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/blink/renderer/core/layout/ng/inline/ng_inline_node.cc b/third_party/blink/renderer/core/layout/ng/inline/ng_inline_node.cc
index 075cd9ebf3b7..c86bcb3a0c3a 100644
--- a/third_party/blink/renderer/core/layout/ng/inline/ng_inline_node.cc
+++ b/third_party/blink/renderer/core/layout/ng/inline/ng_inline_node.cc
@@ -813,7 +813,7 @@ void NGInlineNode::SegmentText(NGInlineNodeData* data) {
 
 // Segment NGInlineItem by script, Emoji, and orientation using RunSegmenter.
 void NGInlineNode::SegmentScriptRuns(NGInlineNodeData* data) {
-  DCHECK_EQ(data->segments, nullptr);
+  DCHECK_EQ(data->segments.get(), nullptr);
 
   String& text_content = data->text_content;
   if (text_content.IsEmpty()) {
-- 
2.25.1

